src/Orleans.Core/Messaging/GatewayConnection.cs

```diff
@@ -85,6 +85,18 @@ public override void Stop()
             CloseSocket(s);
         }
 
+        protected override void Process(Message msg)
+        {
+            if (!IsLive)
+            {
+                RerouteMessage(msg);
+            }
+            else
+            {
+                base.Process(msg);
+            }
+        }
```

Comment:
Note: this logic is in lieu of `DrainQueue`

Reply:
Added comment.